core/modules/taxonomy/taxonomy.admin.inc

```diff
@@ -738,27 +738,24 @@ function taxonomy_form_term($form, &$form_state, TaxonomyTerm $term = NULL, Taxo
   // During initial form build, add the term entity to the form state for use
   // during form building and processing. During a rebuild, use what is in the
   // form state.
+  if (!isset($vocabulary) && isset($term->vocabulary)) {
+    $vocabulary = taxonomy_vocabulary_load($term->vocabulary);
+  }
   if (!isset($form_state['term'])) {
     // Create a new TaxonomyTerm entity for the add form.
     if (!isset($term)) {
-      backdrop_set_title(t('Add term'));
       $term = entity_create('taxonomy_term', array(
         'vocabulary' => $vocabulary->machine_name,
       ));
+      backdrop_set_title(t('Add @vocab_name term', array('@vocab_name' => $vocabulary->name)), PASS_THROUGH);
```

Comment:
```suggestion
      backdrop_set_title(t('Add term to the @vocab_name vocabulary', array('@vocab_name' => $vocabulary->name)), PASS_THROUGH);
```

Reply:
I adjusted using alternate text per the issue discussion.